client.py

```diff
@@ -0,0 +1,344 @@
+        query = self.basic_query()
+        query.update(
+            {
+                "aggregations": {
+                    "daily": {
+                        "date_histogram": {
+                            "field": "publication_date",
+                            "calendar_interval": "day",
+                            "min_doc_count": 1,
+                        }
+                    },
+                    "lang": {"terms": {"field": "language.keyword", "size": 100}},
+                    "domain": {"terms": {"field": "canonical_domain", "size": 100}},
+                    "tld": {"terms": {"field": "tld", "size": 100}},
+                },
+                "track_total_hits": True,
+            }
+        )
+        return query
+
+    def terms_query(self, field):
+        resct = 200
+        aggr_map = {
+            "terms": {
+                "field": field.name,
+                "size": resct,
+                "min_doc_count": 10,
+                "shard_min_doc_count": 5,
+            }
+        }
+        query = self.basic_query()
+        query.update(
+            {
+                "track_total_hits": False,
+                "_source": False,
+                "aggregations": {
+                    "sample": {
+                        "sampler": {"shard_size": 500},
+                        "aggregations": {"topterms": aggr_map},
+                    }
+                },
+            }
+        )
+        return query
+
+    def paged_query(
+        self,
+        resume: Union[str, None],
+        expanded: bool,
+        sort_field=Optional[str],
+        sort_order=Optional[str],
+        page_size=Optional[int],
+    ) -> Dict:
+        query = self.basic_query(expanded)
+        final_sort_field = self._validate_sort_field(sort_field or "publication_date")
+        final_sort_order = self._validate_sort_order(sort_order or "desc")
+        query.update(
+            {
+                "size": self._validate_page_size(page_size or 1000),
+                "track_total_hits": False,
+                "sort": {
+                    final_sort_field: {
+                        "order": final_sort_order,
+                        "format": "basic_date_time_no_millis",
+                    }
+                },
+            }
+        )
+        if resume:
+            # important to use `search_after` instead of 'from' for memory reasons related to paging through more
+            # than 10k results
+            query["search_after"] = [decode_key(resume)]
+        return query
+
+    def article_query(self):
+        default: dict = {
+            "_source": self._expanded_source,
+            "query": {"match": {"_id": self.query_text}},
+        }
+
+        return default
+
+
+class EsClientWrapper:
+    # A wrapper to actually make the calls to elasticsearch
+    def __init__(self, eshosts, **esopts):
+        self.ES = Elasticsearch(eshosts, **esopts)
+        self.maxpage = os.getenv("MAXPAGE", 1000)
```

Comment:
use `int(os.getenv("MAXPAGE", 1000))` ??

Reply:
addressed the typing issue via pydantic_settings!